docs/guide/quickstart.rst

```diff
@@ -0,0 +1,292 @@
+
+We import ``discord.ext.commands`` here as we will need it for our bot.
+
+.. code-block:: python
+
+    intents = discord.Intents.default()
+    bot = commands.Bot(command_prefix="!", description="This is my awesome bot!", intents=intents)
+
+This is your bot instance. You can name the variable whatever you like but for simplicity's sake we will name it ``bot``.
+You will need to specify a ``command_prefix`` here, we use ``!``, but you can use any string, or a list of strings for multiple prefixes.
+
+.. note::
+
+    As of 2.0, the ``intents`` argument is required.
+    ``Intents`` are the method to specify which gateway events you wish to receive. ``Intents.default()`` means you will receive all events that are not locked behind bot verification.
+    For more information, see :ref:`intents_primer`.
+
+.. code-block:: python
+
+    @bot.listen()
+    async def on_ready():
+        print(f"Ready! I am {bot.user} and my ID is {bot.user.id}")
+
+This is the ``ready`` event. It is called when the bot has finished loading and everything is cached.
+We use the :meth:`@bot.listen() <ext.commands.Bot.listen>` decorator as to not override the main event.
+For a list of available events, see :ref:`discord-api-events`.
+
+.. _guide_quickstart_bot_event_warning:
+
+.. warning::
+
+    Using :meth:`@bot.event <ext.commands.Bot.event>` is also valid, but keep in mind that using :meth:`@bot.event <ext.commands.Bot.event>` with the ``on_message`` event can cause your commands to stop responding.
+    To remedy this, either switch to :meth:`@bot.listen() <ext.commands.Bot.listen>`, or add :meth:`~ext.commands.Bot.process_commands` to your ``on_message`` event:
+
+    .. code-block:: python
+
+        @bot.event
+        async def on_message(message: discord.Message):
+            print(f'Received "{message.clean_content}" from {message.author}')
+            # IMPORTANT:
+            await bot.process_commands(message)
+
+.. warning::
+
+    ``on_ready`` can and will be called multiple times throughout your bot's uptime. You should avoid doing any kind of state-management here, such as connecting and loading your database.
+
+.. code-block:: python
+
+    @bot.command()
+    async def echo(ctx: commands.Context, *, sentence: commands.clean_content):
+        await ctx.send(sentence)
+
+Here's where our commands will be defined. We use the :meth:`@bot.command() <ext.commands.Bot.command>` decorator to flag this function as a command.
+This creates a command ``!echo`` that we can type into a channel, and the bot will respond with the given sentence. A few key features:
+
+- :class:`~ext.commands.Context` refers to the command invocation context - this includes the :attr:`~ext.commands.Context.channel`, command :attr:`~ext.commands.Context.author`, :attr:`~ext.commands.Context.message` and more.
+- The ``*`` is a sign to tell discord.py that the following parameter should **consume all text afterward** and condense it into that parameter.
```

Comment:
Perhaps a controversial opinion but should we flag the "dpy specific" features and their comparisons to regular python?
Such as `*` denoting our consumption argument, but in python it makes it a key-word argument, etc etc.
Probably out of our scope but this happens a lot...